src/MongoBson.pas

```diff
@@ -1484,14 +1484,14 @@ function TBsonBuffer.finishObject: Boolean;
 begin
   child := FSubNativeBson.Pop;
   Result := bson_append_document_end(GetCurrNativeBson, child);
-  bson_destroy(child);
+  FreeMem(child);
```

Comment:
@eugene-convey , here rather than delphi's FreeMem() you can use bson_free()

https://github.com/Convey-Compliance/libbson/blob/development/src/bson/bson-memory.c#L213-L217